enhancements/installer/single-node-installation-bootstrap-in-place.md

```diff
@@ -0,0 +1,483 @@
+which will get the master Ignition as input and will enrich the master
+Ignition with control plane static pods manifests and all required
+resources, including the etcd database.
+
+`cluster-bootstrap` normally waits for a list of required pods to be
+ready. These pods are expected to start running on the control plane
+nodes when the bootstrap and control plane run in parallel. That is
+not possible when bootstrapping in place, so when `cluster-bootstrap`
+runs with the `--bootstrap-in-place` option it should only apply the
+manifests and then tear down the control plane.
+
+If `cluster-bootstrap` fails to apply some of the manifests, it should
+return an error.
+
+#### Bootstrap / Control plane static pods
+
+We will review the list of revisions for apiserver/etcd and see if we
+can reduce them by reducing revisions caused by observations of known
+conditions.  For example in a single node we know what the etcd
+endpoints will be in advance, We can avoid a revision by observing
+this post install.  This work will go a long way to reducing
+disruption during install and improve MTTR for upgrade re-deployments
+and failures.
+
+The control plane components we will copy from
+`/etc/kubernetes/manifests` into the master Ignition are:
+
+1. etcd-pod
+2. kube-apiserver-pod
+3. kube-controller-manager-pod
+4. kube-scheduler-pod
+
+These components also require other files generated during bootstrapping:
+
+1. `/var/lib/etcd`
+2. `/etc/kubernetes/bootstrap-configs`
+3. `/opt/openshift/tls/*` (`/etc/kubernetes/bootstrap-secrets`)
+4. `/opt/openshift/auth/kubeconfig-loopback` (`/etc/kubernetes/bootstrap-secrets/kubeconfig`)
+
+**Note**: `/etc/kubernetes/bootstrap-secrets` and `/etc/kubernetes/bootstrap-configs` will be deleted by the `post-reboot` service, after the node reboots (see below).
+
+The bootstrap static pods will be generated in a way that the control
+plane operators will be able to identify them and either continue in a
+controlled way for the next revision, or just keep them as the correct
+revision and reuse them.
+
+#### Post-reboot service
+
+We will add a new `post-reboot` service for approving the kubelet and
+the node Certificate Signing Requests. This service will also cleanup
+the bootstrap static pods resources when the OpenShift control plane
+is ready.
+
+Since we start with a liveCD, the bootstrap services (`bootkube`, `approve-csr`, etc.), `/etc` and `/opt/openshift` temporary files are written to the ephemeral filesystem of the live image, and not to the node's real filesystem.
+
+The files that we need to delete are under:
+
+* `/etc/kubernetes/bootstrap-secrets`
+* `/etc/kubernetes/bootstrap-configs`
+
+These files are required for the bootstrap control plane to start before it is replaced by the control plane operators.
+Once the OCP control plane static pods are deployed we can delete the files as they are no longer required.
+
+#### Requirements for a Single Node deployment with bootstrap-in-place
```

Comment:
"Requirements" can be a bit overloaded, especially in this repository. These are prerequisites that the user has to provide before the deployment, right?

```suggestion
#### Prerequisites for a Single Node deployment with bootstrap-in-place
```